File: leetcode/2024_6_june/python/14_minimum_increment_to_make_array_unique.py
# CODE

from typing import List
class Solution:
    def minIncrementForUnique(self, nums: List[int]) -> int:
        n = len(nums)
        res = 0
        nums.sort()
        for i in range(1, n):
            if nums[i] <= nums[i - 1]:
                inc = nums[i - 1] - nums[i] + 1
                nums[i] = nums[i - 1] + 1
                res += inc
        return res

File: leetcode/2024_6_june/c++/14_minimum_increment_to_make_array_unique.cpp
// TIME COMPLEXITY
// O(nlogn)

// SPACE COMPLEXITY
// O(1)

// CODE

class Solution {
public:
    int minIncrementForUnique(vector<int>& nums) {
        int n = nums.size();
        sort(nums.begin(), nums.end());
        int cnt = 0;
        for(int i = 1; i < n; i++){
            if(nums[i] <= nums[i-1]){
                cnt += abs(nums[i] - nums[i-1]) + 1;
                nums[i] = nums[i-1] + 1;
            }
        }
        return cnt;
    }
};